plugins/common.go

```diff
@@ -0,0 +1,16 @@
+package plugins
+
+import (
+	"github.com/gluster/glusterd2/servers/rest/route"
+	"github.com/prashanthpai/sunrpc"
+)
+
+// GlusterdPlugin is an interface that every Glusterd plugin will
+// implement to add sunrpc program, REST routes and Transaction step
+// functions
+type GlusterdPlugin interface {
+	Name() string
+	SunRPCProgram() sunrpc.Program
+	RestRoutes() route.Routes
+	RegisterStepFuncs()
```

Comment:
I don't know why I had not noticed this before.
`RestRoutes()` and `RegisterStepFuncs()` make up the `Command` interface. You could have just embedded it in the plugin interface.

```go
import "github.com/gluster/glusterd2/commands"

type GlusterPlugin interface {
  Name() string
  SunRPCProgram() sunrpc.Program
  commands.Command
}
```